src/ss_ros.py

```diff
@@ -300,15 +303,21 @@ def wait_for_response(self, response, timeout):
         while datetime.datetime.now() - start_time < timeout:
             time.sleep(0.1)
             # check periodically whether we've received the response we
-            # were waiting for, and if so, we're done waiting 
+            # were waiting for, and if so, we're done waiting
+            if (self.robot_doing_action == None) or (self.robot_speaking == None): #cmhuang: waiting for the subscriber threads to run up
```

Comment:
Please put comments on their own line above the code they refer to, and make sure lines wrap at 80 chars.